websocket/server.go

```diff
@@ -21,8 +21,21 @@ var systemTopic = TopicDescriptor{
 	Type: TopicTypeSystem,
 }
 
+type Server interface {
+	IsClosed() bool
+	Close()
+	TopicIsRegistered(TopicDescriptor) bool
+	RegisterClient(*Client)
+	UnregisterClient(*Client)
+	BroadcastMessage(MessageForBroadcast) error
+	Run(context.Context)
+}
+
+// Compile time verification that *server implements the Server interface
+var _ Server = (*server)(nil)
```

Comment:
https://go.dev/doc/effective_go#interfaces - Scroll to **Interface checks** section

Reply:
Love these things.